dbms/src/Common/tests/gtest_mpmc_queue.cpp

```diff
@@ -688,49 +689,92 @@ CATCH
+        size_t actual_max_size = 5;
+        auxiliary_memory_bound = sizeof(Int64) * actual_max_size;
+        MPMCQueue<Int64> queue(max_size, auxiliary_memory_bound, [](const Int64 &) { return sizeof(Int64); });
+        for (size_t i = 0; i < actual_max_size; i++)
+            ASSERT_TRUE(queue.tryPush(i) == MPMCQueueResult::OK);
+        ASSERT_TRUE(queue.tryPush(actual_max_size) == MPMCQueueResult::FULL);
+        /// after pop one element, the queue can be pushed again
+        ASSERT_TRUE(queue.tryPop(value) == MPMCQueueResult::OK);
+        ASSERT_TRUE(queue.tryPush(actual_max_size) == MPMCQueueResult::OK);
+    }
+
+    {
+        /// case 3: less capacity bound than the auxiliary memory bound
+        auxiliary_memory_bound = sizeof(Int64) * (max_size * 10);
+        MPMCQueue<Int64> queue(max_size, auxiliary_memory_bound, [](const Int64 &) { return sizeof(Int64); });
+        for (size_t i = 0; i < max_size; i++)
+            ASSERT_TRUE(queue.tryPush(i) == MPMCQueueResult::OK);
+        ASSERT_TRUE(queue.tryPush(max_size) == MPMCQueueResult::FULL);
+    }
+
+    {
+        /// case 4, auxiliary memory bound <= 0 means unbounded for auxiliary memory usage
+        std::vector<Int64> bounds{0, -1};
+        for (const auto & bound : bounds)
+        {
+            MPMCQueue<Int64> queue(max_size, bound, [](const Int64 &) { return 1024 * 1024; });
+            for (size_t i = 0; i < max_size; i++)
+                ASSERT_TRUE(queue.tryPush(i) == MPMCQueueResult::OK);
+            ASSERT_TRUE(queue.tryPush(max_size) == MPMCQueueResult::FULL);
+        }
+    }
+
+    {
+        /// case 5 even if the element's auxiliary memory is out of bound, at least one element can be pushed
+        MPMCQueue<Int64> queue(max_size, 1, [](const Int64 &) { return 10; });
+        ASSERT_TRUE(queue.tryPush(1) == MPMCQueueResult::OK);
+        ASSERT_TRUE(queue.tryPush(2) == MPMCQueueResult::FULL);
+        ASSERT_TRUE(queue.tryPop(value) == MPMCQueueResult::OK);
+        ASSERT_TRUE(queue.tryPop(value) == MPMCQueueResult::EMPTY);
+        ASSERT_TRUE(queue.tryPush(1) == MPMCQueueResult::OK);
+    }
+
+    {
+        /// case 6 after pop a huge element, more than one small push can be notified without further pop
+        MPMCQueue<Int64> queue(max_size, 20, [](const Int64 & element) { return std::abs(element); });
+        ASSERT_TRUE(queue.tryPush(100) == MPMCQueueResult::OK);
+        ASSERT_TRUE(queue.tryPush(5) == MPMCQueueResult::FULL);
+        auto thread_manager = newThreadManager();
+        thread_manager->schedule(false, "thread_1", [&]() {
+            queue.push(5);
+        });
+        thread_manager->schedule(false, "thread_2", [&]() {
+            queue.push(6);
+        });
+        std::exception_ptr current_exception = nullptr;
+        try
+        {
+            std::this_thread::sleep_for(std::chrono::seconds(2));
+            ASSERT_TRUE(queue.pop(value) == MPMCQueueResult::OK);
```

Comment:
```suggestion
            ASSERT_TRUE(queue.popTimeout(value, xxxx) == MPMCQueueResult::OK);
```
Avoid permanent blocking in test

Reply:
This pop is expected to pop the first element(100) from the queue, since we already ensure it is already pushed into the queue successfully in L751, I think it is safe to use `pop()` here.